packages/dashboard-core-plugins/src/panels/ChartPanel.tsx

```diff
@@ -684,10 +685,17 @@ export class ChartPanel extends Component<ChartPanelProps, ChartPanelState> {
     });
   }
 
-  handleError(): void {
+  handleError(error: Error): void {
     // Don't want to set an error state, because the user can fix a chart error within the chart itself.
     // We're not loading anymore either so stop showing the spinner so the user can actually click those buttons.
-    this.setState({ isLoading: false });
+    this.setState({ isLoading: false, error: error.message });
+  }
```

Comment:
Take a look at the comment there:

> Don't want to set an error state, because the user can fix a chart error within the chart itself.

With this change, you're wiring up an error dismissal mechanism through `Panel` for this feature. The errors presented in `Panel` are for when there's an issue displaying the panel or your connection (see in `WidgetPanel`) - they're not really errors you can dismiss, as you won't be able to do anything.
Instead, I'd make a `ChartErrorOverlay` in the `chart` package for the overlay, then in `Chart.tsx` display the error overlay when appropriate, and still emitting the `onError`. No need to wire this through `WidgetPanel` or `Panel`, that would require some more architecture.